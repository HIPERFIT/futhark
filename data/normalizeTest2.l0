fun int main(int a) = 
    let y = let zz = 2*a in replicate(zz,a) in 
    let bound = let x1 = 2*y[2*a-a-a] in 2*x1 in
    loop (res=a) = for i < bound do
        let z = (let y[0] = 
                    (let x = 5*a-7 in a*x - 3*a*y[let q = x/3 in q-a]) 
                    in a* reduce( op +, a+reduce( op +, 0, map(op +(1), y) ), let qq = 2*a in map(fn int (int q) => qq+q, y) )
                ) 
        in z*a 
    in res*2


