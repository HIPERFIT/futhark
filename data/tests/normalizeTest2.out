Result of evaluation: 72754416
